app/controllers/cars_controller.rb

```diff
@@ -0,0 +1,52 @@
+class CarsController < ApplicationController
+  before_action :set_car, only: %i[show update destroy]
+
+  # GET /cars
+  def index
+    @cars = Car.all
+
+    render json: @cars
+  end
+
+  # GET /cars/1
+  def show
+    render json: @car
+  end
+
+  # POST /cars
+  def create
+    @car = Car.new(car_params)
+
+    if @car.save
+      render json: @car, status: :created, location: @car
+    else
+      render json: @car.errors, status: :unprocessable_entity
+    end
+  end
+
+  # PATCH/PUT /cars/1
+  def update
+    if @car.update(car_params)
+      render json: @car
+    else
+      render json: @car.errors, status: :unprocessable_entity
+    end
+  end
+
+  # DELETE /cars/1
+  def destroy
+    @car.destroy!
+  end
```

Comment:
Please consider for the event where the Car is not found or the user is not authorized to do so. In any case where item could not be destroyed, like you have for the other actions. Please add an `if else` block to consider the event where the action is failed

Reply:
# Is it like this?
def destroy
  if @car.destroy
    redirect_to cars_path, notice: 'Car was successfully destroyed.'
  else
    redirect_to cars_path, alert: 'Failed to destroy the car.'
  end
end

Reply:
No @SAZZAD1Q2, we don't redirect a user from the API. That is for server-side rendering when have views on our rails app to redirect the user to. We don't know about any URLs on API to redirect the user.

```suggestion
  # DELETE /cars/1
  def destroy
    if @car.destroy
      head :no_content
    else
      render json: { errors: @car.errors.full_messages.to_sentence }, status: :unprocessable_entity
  end
```